src/main/java/org/icgc/argo/program_service/services/ProgramService.java

```diff
@@ -1,488 +1,460 @@
+import static org.icgc.argo.program_service.model.join.ProgramCountry.createProgramCountry;
+import static org.icgc.argo.program_service.model.join.ProgramInstitution.createProgramInstitution;
+import static org.icgc.argo.program_service.model.join.ProgramPrimarySite.createProgramPrimarySite;
+import static org.icgc.argo.program_service.utils.CollectionUtils.*;
+import static org.icgc.argo.program_service.utils.CollectionUtils.mapToSet;
+import static org.icgc.argo.program_service.utils.EntityService.*;
+
+import io.grpc.Status;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import javax.validation.ValidatorFactory;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.icgc.argo.program_service.converter.ProgramConverter;
+import org.icgc.argo.program_service.model.entity.*;
+import org.icgc.argo.program_service.model.exceptions.NotFoundException;
+import org.icgc.argo.program_service.model.join.*;
+import org.icgc.argo.program_service.proto.Program;
+import org.icgc.argo.program_service.repositories.*;
+import org.icgc.argo.program_service.repositories.query.ProgramSpecificationBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+
+@Service
+@Validated
+@Slf4j
+public class ProgramService {
+
+  /** Dependencies */
+  private final ProgramRepository programRepository;
+
+  private final CancerRepository cancerRepository;
+  private final PrimarySiteRepository primarySiteRepository;
+  private final InstitutionRepository institutionRepository;
+  private final RegionRepository regionRepository;
+  private final CountryRepository countryRepository;
+  private final ProgramConverter programConverter;
+  private final ProgramCancerRepository programCancerRepository;
+  private final ProgramPrimarySiteRepository programPrimarySiteRepository;
+  private final ProgramInstitutionRepository programInstitutionRepository;
+  private final ProgramRegionRepository programRegionRepository;
+  private final ProgramCountryRepository programCountryRepository;
+  private final ValidatorFactory validatorFactory;
+
+  @Autowired
+  public ProgramService(
+      @NonNull ProgramRepository programRepository,
+      @NonNull CancerRepository cancerRepository,
+      @NonNull PrimarySiteRepository primarySiteRepository,
+      @NonNull ProgramConverter programConverter,
+      @NonNull ProgramCancerRepository programCancerRepository,
+      @NonNull ProgramPrimarySiteRepository programPrimarySiteRepository,
+      @NonNull InstitutionRepository institutionRepository,
+      @NonNull RegionRepository regionRepository,

```

Comment:
as we are removing region, is this still needed?

Reply:
These are still being used through grpc endpoints for the getRegions RPC